src/firebase/auth/functions/index.js

```diff
@@ -144,7 +160,21 @@ async function checkIfUserExists(data) {
 exports.sendWelcomeEmail = functions.auth.user().onCreate((user) => {
   if (user.email !== null) {
     const email = user.email;
-    return sendWelcomeEmail(email);
+    try {
+      setTimeout(async () => {
+        const userRef = firestoreDb.collection("airqo-app-users")
+            .doc(user.uid);
+        const userDoc = await userRef.get();
+
+        let firstName = userDoc.data().firstName;
+        if (firstName == null) {
+          firstName = "";
+        }
+        sendWelcomeEmail(email, firstName);
+      }, 300000);
```

Comment:
@BenjaminSsempala , first take me through the purpose of the purpose of this setTimeout functionality....why did we need it? Thanks!

Reply:
The set timeout gives the user time to finish setting up their account and input their account details, i.e. the name right after the signup process. We use the name of the user in the welcome email.